packages/api/src/util/copyCropPlan.js

```diff
@@ -0,0 +1,639 @@
+        const task = _omit(pmp.plant_task.task, [
+          ...Object.keys(BaseModel.baseProperties),
+          'locationTasks',
+        ]);
+        const plantTask = _omit(pmp.plant_task, ['task']);
+        tasks.push({ ...task, plant_task: plantTask });
+        delete pmp.plant_task;
+      }
+      if (pmp.transplant_task) {
+        const task = _omit(pmp.transplant_task.task, [
+          ...Object.keys(BaseModel.baseProperties),
+          'locationTasks',
+        ]);
+        const transplantTask = _omit(pmp.transplant_task, ['task']);
+        tasks.push({ ...task, transplant_task: transplantTask });
+        delete pmp.transplant_task;
+      }
+      pmp.managementTasks.forEach((mt) => {
+        tasks.push(_omit(mt.task, [...Object.keys(BaseModel.baseProperties), 'locationTasks']));
+      });
+      delete pmp.managementTasks;
+    });
+
+    return {
+      management_plan: {
+        ..._omit(mp, Object.keys(BaseModel.baseProperties)),
+        management_plan_group: {
+          management_plan_group_id: mpg.management_plan_group_id,
+          repetition_count: mpg.repetition_count,
+        },
+      },
+      tasks,
+    };
+  });
+};
+
+/**
+ * Retrieves the properties to delete from the Model based on its template mapping schema.
+ *
+ * @param {Object} Model - The model object for which properties to delete will be determined.
+ * @returns {string[]} - An array of property names that are marked for deletion in the Model's template mapping schema.
+ */
+export const getPropertiesToDelete = (Model) => {
+  // Check if the Model has a templateMappingSchema property and get the keys.
+  const propertyAndRelationKeys = Model.templateMappingSchema
+    ? Object.keys(Model.templateMappingSchema)
+    : [];
+
+  // If there are properties in the template mapping schema, filter those marked for deletion ('omit' or 'edit').
+  return propertyAndRelationKeys.length
+    ? propertyAndRelationKeys.filter((key) =>
+        ['omit', 'edit'].includes(Model.templateMappingSchema[key]),
+      )
+    : [];
+};
+
+/**
+ * Retrieves an array of dates from the provided managementPlanGraph object.
+ *
+ * The function chooses a dates from each task within the managementPlanGraph
+ * and returns them in a single array.
+ *
+ * @param {Object} managementPlanGraph - The management plan graph object.
+ * @returns {string[]} - An array of date strings extracted from various tasks within the management plan graph.
+ */
+export const getDatesFromManagementPlanGraph = (managementPlanGraph) => {
+  const dates = [];
+  managementPlanGraph.crop_management_plan.planting_management_plans.forEach((plan) => {
+    // Check if there are plant tasks, transplant tasks or management tasks
+    // and add their complete_date or due_date to the dates array.
+    if (plan.plant_task) {
+      plan.plant_task.task.complete_date
+        ? dates.push(plan.plant_task.task.complete_date)
+        : dates.push(plan.plant_task.task.due_date);
```

Comment:
Just as another option to do the same thing:
```
dates.push(plan.plant_task.task.complete_date || plan.plant_task.task.due_date)
```

Reply:
I am still learning the best use of ternary vs null coalescing vs || &&. But I think an issue is cause by switching to the || here.

The test fails when the change is made indicating that the incorrect date may be pushed. I think because it is doing an array action it will "successfully" push a null complete_date.